src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -301,65 +295,38 @@ public void updateFilteredRoomList(Predicate<Room> predicate) {
     @Override
     public Optional<Task> getTaskFromRoomWithTaskIndex(Index taskIndex, Room room) {
         requireAllNonNull(taskIndex, room);
-        List<Task> tasks = room.getTaskList().asUnmodifiableObservableList();
-        if (taskIndex.getZeroBased() >= tasks.size()) {
-            return Optional.empty();
-        }
-        return Optional.of(tasks.get(taskIndex.getZeroBased()));
+        return room.getTaskWithTaskIndex(taskIndex);
     }
 
     @Override
     public void addTaskToRoom(Task task, Room room) {
         requireAllNonNull(task, room);
-        roomList.addTaskToRoom(task, room);
-    }
-
-    @Override
-    public void addTask(Task task) {
-        requireNonNull(task);
-        taskList.add(task);
-        filteredTasks.setPredicate(PREDICATE_SHOW_ALL_TASKS);
+        assert roomList.containsRoom(room) : "Room must be one of the rooms in the RoomList.";
```

Comment:
will this throw an `Assertion exception` if the roomlist does not contain the room or is there a check to ensure that the roomlist contains the room before calling this method?

Reply:
It will throw an `AssertionError` when `RoomList` does not contain room. The method actually stills work without the assertion. That said, when we are calling from `logic`, the room passed to this method should be in the `RoomList`. The assertion warns the developers.